src/schema/opportunity.ts

```diff
@@ -464,6 +711,109 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         keywords: [],
       };
     },
+    getOpportunities: async (
+      _,
+      args: ConnectionArguments & { state?: number },
+      ctx: Context,
+      info,
+    ) => {
+      // Default to LIVE opportunities if no state is provided
+      const opportunityState = args.state ?? OpportunityState.LIVE;
+
+      if (!ctx.userId) {
+        throw new NotFoundError('Not found!');
+      }
+
+      return await queryPaginatedByDate<
+        GQLOpportunity,
+        'createdAt',
+        typeof args
+      >(
+        ctx,
+        info,
+        args,
+        { key: 'createdAt', maxSize: 50 },
+        {
+          queryBuilder: (builder) => {
+            builder.queryBuilder.where({ state: opportunityState });
+
+            if (!ctx.isTeamMember) {
+              builder.queryBuilder
+                .innerJoin(
+                  'opportunity_user',
+                  'ou',
+                  `ou.opportunityId = ${builder.alias}.id`,
+                )
+                .andWhere('ou.userId = :userId', { userId: ctx.userId })
+                .andWhere('ou.type = :type', {
+                  type: OpportunityUserType.Recruiter,
+                });
+            }
+
+            return builder;
+          },
+          orderByKey: 'DESC',
+          readReplica: true,
+        },
+      );
+    },
+    getOpportunityMatches: async (
+      _,
+      args: ConnectionArguments & { opportunityId: string },
+      ctx: AuthContext,
+      info,
+    ) => {
+      if (!ctx.userId) {
+        throw new NotFoundError('Not found!');
+      }
+
+      // First verify the user has access to this opportunity
+      await ensureOpportunityPermissions({
+        con: ctx.con.manager,
+        userId: ctx.userId,
+        opportunityId: args.opportunityId,
+        permission: OpportunityPermissions.ViewDraft,
+      });
+
+      return await queryPaginatedByDate<
+        GQLOpportunityMatch,
+        'createdAt',
+        typeof args
+      >(
+        ctx,
+        info,
+        args,
+        { key: 'createdAt', maxSize: 50 },
+        {
+          queryBuilder: (builder) => {
+            builder.queryBuilder
+              .where({ opportunityId: args.opportunityId })
+              .andWhere(`${builder.alias}.status IN (:...statuses)`, {
+                statuses: [
+                  OpportunityMatchStatus.CandidateAccepted,
+                  OpportunityMatchStatus.RecruiterAccepted,
+                  OpportunityMatchStatus.RecruiterRejected,
+                ],
+              })
+              // Order by candidate_accepted status first (priority 0), then others (priority 1)
+              // Then by createdAt ascending (oldest first) within each group
+              .addOrderBy(
+                `CASE WHEN ${builder.alias}.status = :candidateAcceptedStatus THEN 0 ELSE 1 END`,
+                'ASC',
+              )
+              .addOrderBy(`${builder.alias}.createdAt`, 'ASC')
```

Comment:
`updatedAt` is the right one. it will order by the matches according to the time it was shown to the recruiter, not when it was sent to the candidate